src/main/java/org/truffleruby/core/module/ModuleOperations.java

```diff
@@ -569,6 +571,10 @@ private static SuperMethodLookup lookupSuperMethod(DynamicObject declaringModule
         return lookup.withResult(new MethodLookupResult(null, toArray(assumptions)));
     }
 
+    private static InternalMethod rememberUsedRefinement(InternalMethod method, DeclarationContext declarationContext) {
```

Comment:
Maybe rememberUsedRefinement*s* since there might be multiple active refinements?